chaoskube/chaoskube_test.go

```diff
@@ -93,143 +80,6 @@ func (suite *Suite) TestRunContextCanceled() {
-		{"", "", "!default,!testing", []map[string]string{}},
-		{"", "", "default,!testing", []map[string]string{foo}},
-		{"", "", "default,!default", []map[string]string{}},
-	} {
-		labelSelector, err := labels.Parse(tt.labelSelector)
-		suite.Require().NoError(err)
-
-		annotationSelector, err := labels.Parse(tt.annotationSelector)
-		suite.Require().NoError(err)
-
-		namespaceSelector, err := labels.Parse(tt.namespaceSelector)
-		suite.Require().NoError(err)
-
-		chaoskube := suite.setupWithPods(
-			labelSelector,
-			annotationSelector,
-			namespaceSelector,
-			[]time.Weekday{},
-			[]util.TimePeriod{},
-			[]time.Time{},
-			time.UTC,
-			time.Duration(0),
-			false,
-		)
-
-		suite.assertCandidates(chaoskube, tt.pods)
-	}
-}
-
-func (suite *Suite) TestVictim() {
-	foo := map[string]string{"namespace": "default", "name": "foo"}
-	bar := map[string]string{"namespace": "testing", "name": "bar"}
-
-	for _, tt := range []struct {
-		seed          int64
-		labelSelector string
-		victim        map[string]string
-	}{
-		{2000, "", foo},
-		{4000, "", bar},
-		{4000, "app=foo", foo},
-	} {
-		rand.Seed(tt.seed)
-
-		labelSelector, err := labels.Parse(tt.labelSelector)
-		suite.Require().NoError(err)
-
-		chaoskube := suite.setupWithPods(
-			labelSelector,
-			labels.Everything(),
-			labels.Everything(),
-			[]time.Weekday{},
-			[]util.TimePeriod{},
-			[]time.Time{},
-			time.UTC,
-			time.Duration(0),
-			false,
-		)
-
-		suite.assertVictim(chaoskube, tt.victim)
-	}
-}
-
-// TestNoVictimReturnsError tests that on missing victim it returns a known error
-func (suite *Suite) TestNoVictimReturnsError() {
-	chaoskube := suite.setup(
-		labels.Everything(),
-		labels.Everything(),
-		labels.Everything(),
-		[]time.Weekday{},
-		[]util.TimePeriod{},
-		[]time.Time{},
-		time.UTC,
-		time.Duration(0),
-		false,
-	)
-
-	_, err := chaoskube.Victim()
-	suite.Equal(err, errPodNotFound)
-	suite.EqualError(err, "pod not found")
-}
-
-func (suite *Suite) TestDeletePod() {
-	foo := map[string]string{"namespace": "default", "name": "foo"}
-	bar := map[string]string{"namespace": "testing", "name": "bar"}
-
-	for _, tt := range []struct {
-		dryRun        bool
-		remainingPods []map[string]string
-	}{
-		{false, []map[string]string{bar}},
-		{true, []map[string]string{foo, bar}},
-	} {
-		chaoskube := suite.setupWithPods(
-			labels.Everything(),
-			labels.Everything(),
-			labels.Everything(),
-			[]time.Weekday{},
-			[]util.TimePeriod{},
-			[]time.Time{},
-			time.UTC,
-			time.Duration(0),
-			tt.dryRun,
-		)
-
-		victim := util.NewPod("default", "foo", v1.PodRunning)
-
-		err := chaoskube.ApplyChaos(victim)
-		suite.Require().NoError(err)
-
-		if tt.dryRun {
-			suite.assertLog(log.InfoLevel, "dry run", log.Fields{"namespace": "default", "name": "foo"})
-		} else {
-			suite.assertLog(log.InfoLevel, "terminating pod", log.Fields{"namespace": "default", "name": "foo"})
-		}
-		suite.assertCandidates(chaoskube, tt.remainingPods)
-	}
-}
-
```

Comment:
So the thought here was that these tests are replaced by the tests [here](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R24); the tests deleted here are going via functions I moved into the spec. 

The test coverage is somewhat reduced because the suite here tests a lot of cases of the LabelSelector parsing stuff, like this test for namespace filtering:

```
{"", "", "default,testing", []map[string]string{foo, bar}},
{"", "", "!testing", []map[string]string{foo}},
{"", "", "!default,!testing", []map[string]string{}},
{"", "", "default,!testing", []map[string]string{foo}},
{"", "", "default,!default", []map[string]string{}},
```

Which is replaced with [this single test case](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R61) (and the cases around it that use the `""` filter); my thought here being that if any filtering by namespace works, we should be able to expect k8s LabelSelector parser to cover the myriad of possible filters that can be written already.

My intention was that these would have equivalent cases covered elsewhere. If I walk through them to verify that.. 

-`TestCandidates` tests label, annotation and namespace filters; the empty case, the negative case and the positive case for each. Those all have equivalents [here](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R36)
- `TestVictim` tests that selection eventually chooses all possible instances, that's covered [here](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R94) by the suite asserting this same thing for every case, rather than just a subset
- `TestNoVictimReturnsError` - since Victim does not exist as a method anymore, that specific behavior is gone; instead we test the 0 victims case doesn't crash the monkey [here](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R77) 
- `TestDeletePod` is confusingly named, it's actually testing the dry run action.. I dropped this guy because we cover that an action is invoked, and I thought we covered the actions themselves separately. But I'm realizing there's no separate tests for them, and either way having a higher level test ensuring "dry run" really is dry run seems important. Will add this back in some form
- `TestMinimumAge` arguably has lost coverage; there are more cases covered in the removed test than in the [new test](https://github.com/neo-technology/marmoset/pull/2/files/c1cc245fe31e3a0aceb4a83566fe2c4bcb0caef8#diff-abde39b8b6a41580567f304fe50efa51R66) adding more cases here as well

Beyond that, there's a notable difference: The tests here largely assert on "given this filter, the possible set of candidates should be X". The new tests also do this, but they're testing on a higher level, so it's "given this filter, eventually this set, and only this set, is selected". There's a neatness to the old tests that is a bummer to lose, but I also quite like the uniformity of the new tests..

All-in-all: I will add more coverage for the age filter and coverage for the dry run feature; shout if you see a specific case you'd want to add beyond that!